### PR TITLE
Add fix for Relay/Gravity count discrepencies.

### PR DESCRIPTION
**Pull request by ashfurrow**
Gravity has a bug (see [this PR](https://github.com/artsy/gravity/pull/11240)) where total counts being returned for user favourites include unpublished artworks. This count discrepancy causes Relay to return `hasNextPage` as `true` while `endCursor` is `null`, which causes serious problems for users on Eigen (see [this Sentry report](https://sentry.io/artsynet/eigen/issues/347342766/events/7196937891/)). 

This pull requests overrides the `pageInfo.hasNextPage` value returned from Relay to set it to false if the end cursor is missing. I've tested locally on Eigen to verify the fix works, and have added test coverage.

Serious props to @orta for his help reproducing this bug. 

Also removes `console.log` statements I accidentally'd last night.

### CODE REVIEW

**Comment by alloy** [comment]
Why is this change needed if the issue lies in Gravity?

**Comment by ashfurrow** [comment]
The Gravity fix is non-trivial and has performance implications I wanted to discuss with the platform team; this PR is a workaround until we can sort that out.

**Comment by alloy** [comment]
Let’s not rush this until we hear from platform people that the Gravity fix needs more consideration. /cc @mzikherman

**Comment by alloy** [comment]
Oh, looks like @mzikherman _just_ commented on the Gravity PR that he considers the performance implication ok.

**Comment by alloy** [comment]
In case the Gravity PR doesn’t make it into production soon after all, then I’m ok on merging and deploying this. It looks good to me 👍

File: schema/collection.js [hunk 1]
```diff
@@ -41,13 +42,20 @@ export const CollectionType = new GraphQLObjectType({
...
+            const connection = connectionFromArraySlice(body, options, {
               arrayLength: headers["x-total-count"],
               sliceStart: gravityOptions.offset,
             })
+            // Gravity has a bug (see PR https://github.com/artsy/gravity/pull/11240) where the counts
+            // returned in the X-Total-Count header includes unpublished artworks, even though only
+            // published artworks are returned. This can lead to a situation where hasNextPage is true
+            // but endCursor is null, violating Relay norms and causing serious problems for Eigen.
+            connection.pageInfo.hasNextPage =
+              connection.pageInfo.endCursor === null ? false : connection.pageInfo.hasNextPage
```
Comment (damassi) [inline schema/collection.js]: I would add a `FIXME:` that points to the Gravity PR so that this jumps out in our tooling.

**Comment by ashfurrow** [comment]
Following up, this has been deployed and appears to fix the problem. Thanks again everyone.

**Comment by ashfurrow** [comment]
Confirming that this did in fact fix the eigen crash:

screen shot 2017-09-13 at 5 59 49 pm

I've marked the issue as resolved in Sentry.

**Pull request merged**